src/test/incremental/mir-opt.rs

```diff
@@ -0,0 +1,11 @@
+// MIR optimizations can create expansions after the TyCtxt has been created.
```

Comment:
Is this about `ExpnKind::Inlined`?
It's a recent addition, and I never manage to get to investigating what it does.
Looks like a total misuse of the expansion machinery at first glance.

Reply:
Yes, `ExpnKind::Inlined` is involved in this case. I did not try to remove it: I find the possibility to create expansions after the TyCtxt exists interesting for (eventually) incremental macro expansion.